addons/document/document.py

```diff
@@ -108,12 +108,27 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None, context=None,
         for parent_id in visible_parent_ids:
             ids.extend(parents[parent_id])
 
+        set_ids = set(ids)
+
         # sort result according to the original sort ordering
         if count:
-            return len(ids)
+            result = len(set_ids)
         else:
-            set_ids = set(ids)
-            return [id for id in orig_ids if id in set_ids]
+            result = [id for id in orig_ids if id in set_ids]
```

Comment:
Don't use `id` for variable name, but `rec_id` or similar. Why don't you make `set_id.intersection(set(orig_ids))`?